src/nautilus_librarian/mods/gpg/domain/utils.py

```diff
@@ -0,0 +1,127 @@
+
+    Sample output for the command:
+    sec:-:4096:1:27304EDD6079B81C:1637342753:::-:::scESC:::+:::23::0:
+    fpr:::::::::88966A5B8C01BD04F3DA440427304EDD6079B81C:
+    grp:::::::::449972AC9FF11BCABEED8A7AE834C4349CC4DBFF:
+    uid:-::::1637342753::B3B0B2247600E80BAB9D4802D5CF0AFC477DE016::A committer <committer@example.com>::::::::::0:
+    ssb:-:4096:1:5B6BDD35BEDFBF6F:1637342753::::::e:::+:::23:
+    fpr:::::::::B1D4A2483D1D2A02416BE0775B6BDD35BEDFBF6F:
+    grp:::::::::97D36F5B8F5BECDA8A1923FC00D11C7C438584F9:
+    In that example the keygrip (grp) of the key 88966A5B8C01BD04F3DA440427304EDD6079B81C is
+    449972AC9FF11BCABEED8A7AE834C4349CC4DBFF
+    Specification for the format: https://github.com/CSNW/gnupg/blob/master/doc/DETAILS
+    """
+
+    output = get_key_details_with_colons_format(fingerprint)
+
+    records = output.split("\n")
+
+    current_fingerprint = ""
+
+    for record in records:
+        if record.startswith("fpr"):
+            fields = record.split(":")
+            current_fingerprint = fields[9]
+        if record.startswith("grp"):
+            fields = record.split(":")
+            keygrip = fields[9]
+            if current_fingerprint == fingerprint:
+                return keygrip
+
+    return None
+
+
+def get_key_user_by(fingerprint):
+    """
+    This function gets the uid record of a given GPG key using the gpg console command and parsing the output.
+
+    Sample output for the command:
+    sec:-:4096:1:27304EDD6079B81C:1637342753:::-:::scESC:::+:::23::0:
+    fpr:::::::::88966A5B8C01BD04F3DA440427304EDD6079B81C:
+    grp:::::::::449972AC9FF11BCABEED8A7AE834C4349CC4DBFF:
+    uid:-::::1637342753::B3B0B2247600E80BAB9D4802D5CF0AFC477DE016::A committer <committer@example.com>::::::::::0:
+    ssb:-:4096:1:5B6BDD35BEDFBF6F:1637342753::::::e:::+:::23:
+    fpr:::::::::B1D4A2483D1D2A02416BE0775B6BDD35BEDFBF6F:
+    grp:::::::::97D36F5B8F5BECDA8A1923FC00D11C7C438584F9:
+    In that example the 'uid' of the key 88966A5B8C01BD04F3DA440427304EDD6079B81C is A committer <committer@example.com>
+    There could be more than one 'uid' record. We return the first one.
+    Specification for the format: https://github.com/CSNW/gnupg/blob/master/doc/DETAILS
+    """
+
+    output = get_key_details_with_colons_format(fingerprint)
+
+    records = output.split("\n")
+
+    current_fingerprint = ""
+
+    for record in records:
+        if record.startswith("fpr"):
+            fields = record.split(":")
+            current_fingerprint = fields[9]
+        if record.startswith("uid"):
+            fields = record.split(":")
+            uid = fields[9]
+            if current_fingerprint == fingerprint:
+                name, separator, rest = uid.partition(" <")
+                email, separator, rest = rest.partition(">")
+                return name, email
+
+    return None
+
+
+def get_short_key_id_from_fingerprint(fingerprint):
```

Comment:
Is there are particular reason we need to use `Short Key ID's` or `Long Key ID's`? Both are so short they can be easily ground.